core/src/main/java/org/elasticsearch/index/query/support/InnerHitsQueryParserHelper.java

```diff
@@ -130,4 +132,20 @@ public static void parseCommonInnerHitOptions(XContentParser parser, XContentPar
             }
         }
     }
+
+    private static void parseHighlight(XContentParser parser, SearchContext context) {
+        QueryParseContext parseContext = context.getQueryShardContext( ).parseContext();
+        XContentParser old = parseContext.parser();
```

Comment:
Do you do this because the parse isn't in the context? I thought this kind of manipulation would have been done way up the call stack when we initialize the context.